composeml/label_maker.py

```diff
@@ -89,49 +96,46 @@ def search(self, df, minimum_data, num_examples_per_instance, gap, verbose=True,
         progress_bar = tqdm(total=total, bar_format=bar_format, disable=not verbose, file=stdout)
 
         def df_to_labels(df):
-            labels = pd.Series()
             df = df.loc[df.index.notnull()]
             df.sort_index(inplace=True)
-
-            if df.empty:
-                return labels
-
             cutoff_time = offset_time(df.index, minimum_data)
 
+            labels = pd.Series()
             for example in range(num_examples_per_instance):
-                df = df[cutoff_time:]
-
-                if df.empty:
-                    skipped_iterations = num_examples_per_instance - example
-                    progress_bar.update(n=skipped_iterations)
+                if cutoff_time is None:
+                    n = num_examples_per_instance - example
+                    progress_bar.update(n=n)
                     break
 
-                window_end = offset_time(df.index, self.window_size)
-                label = self.labeling_function(df[:window_end], *args, **kwargs)
+                df = df[cutoff_time:]
+                window = offset_time(df.index, self.window_size)
+                label = self.labeling_function(df[:window], *args, **kwargs)
 
                 if not pd.isnull(label):
                     labels[cutoff_time] = label
 
                 cutoff_time = offset_time(df.index, gap)
                 progress_bar.update(n=1)
 
-            labels.index = labels.index.rename('cutoff_time')
             labels.index = labels.index.astype('datetime64[ns]')
+            labels.index.rename('cutoff_time', inplace=True)
             return labels
 
         labels_per_group = []
         for key, df in groups:
             labels = df_to_labels(df)
-            labels = labels.to_frame(name=name)
-            labels[self.target_entity] = key
-            labels_per_group.append(labels)
+
+            if not labels.empty:
+                labels = labels.to_frame(name=name)
+                labels[self.target_entity] = key
+                labels_per_group.append(labels)
 
         progress_bar.close()
-        labels = pd.concat(labels_per_group, axis=0, sort=False)
 
-        if labels.empty:
+        if len(labels_per_group) == 0:
             return LabelTimes(name=name, target_entity=self.target_entity)
 
+        labels = pd.concat(labels_per_group, axis=0, sort=False)
         labels.reset_index(inplace=True)
         labels.rename_axis('label_id', inplace=True)
 
```

Comment:
any reason not to pass the settings when you init the `LabelTimes` object? why wait and call update on it?